### PR TITLE
#1 - Project setup

### LINKED ISSUE #1: Project setup
Setup of project.

### PR DESCRIPTION
**Pull request by miroslavpojer**
- Initial setup of repository.
- No code included.

Closes #1

### CODE REVIEW

File: LICENSE.LGPL
Comment (Zejnilovic) [inline LICENSE.LGPL:1]: Licence is always an `.MD` file in our repos. Why is this `.LGPL`?

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,65 @@
+name: Build and Test
```
Comment (benedeki) [inline .github/workflows/build.yml]: I believe, it's easier to follow, if the filename and workflow name correspond.
E.g.
`build.yaml` - `build_and_test.yaml`
or
```suggestion
name: Build
```

File: .gitignore [hunk 2]
```diff
@@ -159,4 +159,4 @@ cython_debug/
 #  be found at https://github.com/github/gitignore/blob/main/Global/JetBrains.gitignore
 #  and can be added to the global gitignore or merged into this file.  For a more nuclear
 #  option (not recommended) you can uncomment the following to ignore the entire idea folder.
-#.idea/
+.idea
```
Comment (benedeki) [inline .gitignore:163]: Other projects have also:
```suggestion
# idea
.idea
*.iml
logs

# vs code
.vscode
```

File: .github/ISSUE_TEMPLATE/spike_task.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
+---
+name: Spike
+about: Issue template for spikes, research and investigation tasks
+labels: 'spike'
```
Comment (benedeki) [inline .github/ISSUE_TEMPLATE/spike_task.md:4]: You don't have a `spike` label.
I can give you a script which creates our (my) standard labels, with same colors and descriptions.

File: README.md [hunk 4]
```diff
@@ -1,2 +1,173 @@
...
+- [How to Use Local Version of Library for Development](#how-to-use-local-version-of-library-for-development)
+- [How to Run Tests](#how-to-run-tests)
+- [How to Release](#how-to-release)
+- [Contribution Guidelines](#contribution-guidelines)
+- [License Information](#license-information)
+- [Contact or Support Information](#contact-or-support-information)
+
+
+## Description
+**PyCompositeActionLib** is a Python library that streamlines creating and managing composite GitHub Actions. It offers reusable classes and scripts for controlling action inputs/outputs and accessing the GitHub API.
```
Comment (benedeki) [inline README.md]: **Minor:**
Some long lines all around

File: README.md [hunk 5]
```diff
@@ -1,2 +1,173 @@
...
+
+### License Information
+
+This project is licensed under the Apache License 2.0. It is a liberal license that allows you great freedom in using, modifying, and distributing this software, while also providing an express grant of patent rights from contributors to users.
+
+For more details, see the [LICENSE](https://github.com/AbsaOSS/py-composite-action-lib/blob/master/LICENSE) file in the repository.
+
+### Contact or Support Information
+
+If you need help with using or contributing to Generate Release Notes Action, or if you have any questions or feedback, don't hesitate to reach out:
```
Comment (benedeki) [inline README.md]: Wrong copy&paste?

**Review by benedeki — COMMENTED** [review]
No action to add issues to a project?

File: LICENSE.LGPL
Comment (miroslavpojer) [reply LICENSE.LGPL:1]: I Was following the examples in other GH repos in this situation:
- https://github.com/FFmpeg/FFmpeg/blob/master/COPYING.LGPLv3

I see no problem with adding .md to the file in this repository.

File: .github/ISSUE_TEMPLATE/spike_task.md (hunk 3 above)
Comment (miroslavpojer) [reply .github/ISSUE_TEMPLATE/spike_task.md:4]: Yes, please.

File: README.md (hunk 4 above)
Comment (miroslavpojer) [reply README.md]: What is the pointed problem here?
- Too long line in md file? (final md representation does not care about new lines in example above)
- Reach better human reading?

File: .github/workflows/build.yml (hunk 1 above)
Comment (miroslavpojer) [reply .github/workflows/build.yml]: Addressed in commit 0b351127825f0dd446f0a0828295429ff5627f10 by renaming to test.yml.

File: .gitignore (hunk 2 above)
Comment (miroslavpojer) [reply .gitignore:163]: Thanks. Address in commit 0b351127825f0dd446f0a0828295429ff5627f10.

File: README.md (hunk 5 above)
Comment (miroslavpojer) [reply README.md]: You are right. Fixed in commit - 0b351127825f0dd446f0a0828295429ff5627f10.

File: README.md (hunk 4 above)
Comment (benedeki) [reply README.md]: When rendered by `md` knowledgeable application it's not a problem. When i text editor, it's hard to read though. Afaik, in `md`, if not part of a bullet point, lines are usually line-broken to 80/120 characters.

Comment (miroslavpojer) [reply README.md]: OK, I will apply it. See commit b50390300557815f7ae8e9e6dae0f8884e08c869.

File: LICENSE.LGPL
Comment (miroslavpojer) [reply LICENSE.LGPL:1]: License mentioning rules addressed in commit - 3358c0b1d539226c1bc5b4dc9ea840bbc644c0c0.

File: README.md [hunk 6]
```diff
@@ -1,2 +1,187 @@
...
+set_action_output('output-name', 'success')
+```
+
+### Example: Access GitHub API with GithubManager
+```python
+from github import Github, Auth
+from py_composite_action_lib import GithubManager
+
+# Authenticate to GitHub
+token = 'your_github_token'
```
Comment (Zejnilovic) [inline README.md]: please provide a better example of handling secrets first

File: README.md [hunk 7]
```diff
@@ -1,2 +1,187 @@
...
+# Example: Get action input
+repo_name = get_action_input('repo-name')
+if not repo_name:
+    set_action_failed('The input "repo-name" is required.')
+
+# Example: Set action output
+set_action_output('output-name', 'success')
+```
+
+### Example: Access GitHub API with GithubManager
```
Comment (Zejnilovic) [inline README.md:41]: how do you access ref?

File: README.md (hunk 6 above)
Comment (miroslavpojer) [reply README.md]: Addressed in commit 58ba9758ebbc98cc876c7e5df2285f9a9f20c407.
The action is accessing inputs from env variables.

See the usage of action which is using this library:

```python
- name: Generate Release Notes
  id: release_notes_generator
  uses: AbsaOSS/generate-release-notes@v0.2.0
  env:
    GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}  
  with:
    tag-name: "v0.1.0"
    chapters: '[
      {"title": "Breaking Changes 💥", "label": "breaking-change"},
      {"title": "New Features 🎉", "label": "enhancement"},
      {"title": "New Features 🎉", "label": "feature"},
      {"title": "Bugfixes 🛠", "label": "bug"}
    ]'
    warnings: false
    published-at: true
    skip-release-notes-label: 'ignore-in-release'     # changing default value of label
    print-empty-chapters: false
    chapters-to-pr-without-issue: false
    verbose: false
```

File: README.md (hunk 7 above)
Comment (miroslavpojer) [reply README.md:41]: Access to ref is not needed in planned actions.
- The **release notes generator** scans the entire repository using GH REST API.
- The **living documentation generator** scans multiple repositories and projects using GH REST API.
- The **Filename inspector** - scans the current filesystem using provided paths and glob logic.

**Comment by miroslavpojer** [comment]
Development of the library has been ended.

**Pull request closed without merging**